packages/api/src/user.js

```diff
@@ -187,30 +187,62 @@ export async function userUploadsGet (request, env) {
     size = parsedSize
   }
 
-  let before = new Date()
+  let offset = 0
+  if (searchParams.has('offset')) {
+    const parsedOffset = parseInt(searchParams.get('offset'))
+    if (isNaN(parsedOffset) || parsedOffset <= 0) {
+      throw Object.assign(new Error('invalid page offset'), { status: 400 })
+    }
+    offset = parsedOffset
+  }
+
+  let before
   if (searchParams.has('before')) {
     const parsedBefore = new Date(searchParams.get('before'))
     if (isNaN(parsedBefore.getTime())) {
       throw Object.assign(new Error('invalid before date'), { status: 400 })
     }
-    before = parsedBefore
+    before = parsedBefore.toISOString()
+  }
+
+  let after
+  if (searchParams.has('after')) {
+    const parsedAfter = new Date(searchParams.get('after'))
+    if (isNaN(parsedAfter.getTime())) {
+      throw Object.assign(new Error('invalid after date'), { status: 400 })
+    }
+    after = parsedAfter.toISOString()
   }
 
   const sortBy = searchParams.get('sortBy') || 'Date'
   const sortOrder = searchParams.get('sortOrder') || 'Desc'
 
-  const uploads = await env.db.listUploads(request.auth.user._id, {
+  const data = await env.db.listUploads(request.auth.user._id, {
     size,
-    before: before.toISOString(),
+    offset,
+    before,
+    after,
     sortBy,
     sortOrder
   })
 
-  const oldest = uploads[uploads.length - 1]
-  const headers = uploads.length === size
-    ? { Link: `<${requestUrl.pathname}?size=${size}&before=${encodeURIComponent(oldest.created)}>; rel="next"` }
-    : undefined
-  return new JSONResponse(uploads, { headers })
+  const headers = {
+    Count: data.count,
+    Size: size,
+    Offset: offset
+  }
+
+  if (data.uploads.length + offset < data.count) {
+    const nextOffset = offset + size
+    headers.Next_link = `<${requestUrl.pathname}?size=${size}&offset=${encodeURIComponent(nextOffset)}>; rel="next"`
+  }
+
+  if (offset) {
+    const previousOffset = offset - size
+    headers.Prev_link = `<${requestUrl.pathname}?size=${size}&offset=${encodeURIComponent(previousOffset)}>; rel="next"`
```

Comment:
Thoughts of including this? 
Since we have a next link it seemed to make sense to also include the previous link.
Though I can see an argument for not including links at all.